pkg/controller/azvolume.go

```diff
@@ -77,11 +78,20 @@ func (r *reconcileAzVolume) Reconcile(ctx context.Context, request reconcile.Req
 			//If delete failed, requeue request
 			return reconcile.Result{Requeue: true}, err
 		}
+		// azVolume released, so clean up all attachments
+	} else if azVolume.Status.Phase == v1alpha1.VolumeReleased {
+		klog.Infof("Volume released: Initiating AzVolumeAttachment Clean-up")
+		if err := CleanUpAzVolumeAttachment(ctx, r.client, r.azVolumeClient, r.namespace, azVolume.Name); err != nil {
+			return reconcile.Result{Requeue: true}, err
+		}
+		if err := r.UpdateStatus(ctx, azVolume.Name, v1alpha1.VolumeAvailable, false, azVolume.Status.ResponseObject); err != nil {
+			klog.Errorf("failed to update status of AzVolume (%s): %v", azVolume.Name, err)
+			return reconcile.Result{Requeue: true}, err
+		}
 	} else {
 		// azVolume update
 		if err := r.triggerUpdate(ctx, azVolume.Name); err != nil {
 			klog.Errorf("failed to update AzVolume (%s): %v", azVolume.Name, err)
-			return reconcile.Result{Requeue: true}, err
```

Comment:
Why remove the requeue here? Shouldn't we retry on failure?

Reply:
Good catch. I must have accidentally deleted the line. I will revert the change :)

Reply:
hmmmm strange. My local commit still doesn't have that line removal.